internal/linkhelpers/linkhelpers.go

```diff
@@ -25,50 +44,176 @@ type ResponseJSON struct {
 	Result string `json:"result"`
 }
 
+func (links *URLLinks) OpenDB() error {
+	if links.Cfg.BasePath == "" {
+		links.DB = nil
+		return errors.New("empty path for database")
+	}
+
+	db, err := sql.Open("pgx", links.Cfg.BasePath)
+
+	if err != nil {
+		return err
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Second)
+	defer cancel()
+
+	_, err = db.ExecContext(ctx, "CREATE TABLE IF NOT EXISTS links (id varchar(255), url varchar(255), cookie varchar(255))")
+	if err != nil {
+		return err
+	}
+
+	ctx, cancel = context.WithTimeout(context.Background(), 1*time.Second)
+	defer cancel()
+	rows, err := db.QueryContext(ctx, "SELECT * FROM links")
+
+	if err != nil {
+		return err
+	}
+
+	defer rows.Close()
+
+	for rows.Next() {
+		var short string
+		var url string
+		var cookie string
+		err = rows.Scan(&short, &url, &cookie)
+		if err != nil {
+			return err
+		}
+		links.Locations[short] = url
+		links.Users[cookie] = append(links.Users[cookie], short)
+	}
+
+	err = rows.Err()
+	if err != nil {
+		return err
+	}
+
+	links.DB = db
+	return nil
+}
+func (links *URLLinks) OpenFile() error {
+	if links.Cfg.StoragePath == "" {
+		return errors.New("empty path for file")
+	}
+
+	file, err := os.OpenFile(links.Cfg.StoragePath, os.O_RDWR|os.O_APPEND|os.O_CREATE|os.O_SYNC, 0777)
+	if err != nil {
+		return err
+	}
+	scanner := bufio.NewScanner(file)
+	i := 1
+	for scanner.Scan() {
+		links.Locations[fmt.Sprint(i)] = scanner.Text()
+		i += 1
```

Comment:
```suggestion
		i++
```
 с питона перешел? ))))))

Reply:
да))))